kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -751,7 +752,7 @@ internal class CodeWriter(
       val imported = mutableMapOf<String, Set<T>>()
       forEach { (simpleName, qualifiedNames) ->
         val canonicalNamesToQualifiedNames = qualifiedNames.associateBy { it.computeCanonicalName() }
-        if (canonicalNamesToQualifiedNames.size == 1 && simpleName !in referencedNames) {
+        if (canonicalNamesToQualifiedNames.size == 1 && simpleName !in referencedNames && simpleName !in imported) {
```

Comment:
My understanding is that the `simpleName !in imported` is needed for the case demonstrated by the `importAliasCollisionWithRegularImportWhenAliasesComputedFirst` test: if there's a generated alias that conflicts with a simple name of a different class, we'll be forced to generate an alias for that simple name, so non-conflicting simple names don't take precedence. I think this is a bit unfortunate, and I wonder if pre-sorting `(simpleName, qualifiedNames)` pairs by the number of `qualifiedNames` would be an easy improvement here? That way we'll first process non-conflicting names and only then get to conflicts that require aliases.

Reply:
Sure, that makes sense. I can make that change - this is also probably better because the imports end up being the same regardless of which appears in the file first.

Reply:
Yep!